src/natcap/invest/sdr/sdr_core.pyx

```diff
@@ -136,51 +125,22 @@ def calculate_sediment_deposition(
 
     for offset_dict in pygeoprocessing.iterblocks(
             (mfd_flow_direction_path, 1), offset_only=True, largest_block=0):
-        win_xsize = offset_dict['win_xsize']
-        win_ysize = offset_dict['win_ysize']
-        xoff = offset_dict['xoff']
-        yoff = offset_dict['yoff']
 
         if ctime(NULL) - last_log_time > 5.0:
             last_log_time = ctime(NULL)
             LOGGER.info('Sediment deposition %.2f%% complete', 100 * (
                 n_pixels_processed / float(n_cols * n_rows)))
 
-        for row_index in range(win_ysize):
-            seed_row = yoff + row_index
-            for col_index in range(win_xsize):
-                seed_col = xoff + col_index
-                # check if this is a good seed pixel ( a local high point)
-                if mfd_flow_direction_raster.get(seed_col, seed_row) == mfd_nodata:
-                    continue
-                seed_pixel = 1
-                # iterate over each of the pixel's neighbors
-                for j in range(8):
-                    # skip if the neighbor is outside the raster bounds
-                    neighbor_row = seed_row + ROW_OFFSETS[j]
-                    if neighbor_row < 0 or neighbor_row >= n_rows:
-                        continue
-                    neighbor_col = seed_col + COL_OFFSETS[j]
-                    if neighbor_col < 0 or neighbor_col >= n_cols:
-                        continue
-                    # skip if the neighbor's flow direction is undefined
-                    neighbor_flow_val = <int>mfd_flow_direction_raster.get(
-                        neighbor_col, neighbor_row)
-                    if neighbor_flow_val == mfd_nodata:
-                        continue
-                    # if the neighbor flows into it, it's not a local high
-                    # point and so can't be a seed pixel
-                    neighbor_flow_weight = (
-                        neighbor_flow_val >> (inflow_offsets[j]*4)) & 0xF
-                    if neighbor_flow_weight > 0:
-                        seed_pixel = 0  # neighbor flows in, not a seed
-                        break
+        for row_index in range(offset_dict['win_ysize']):
+            ys = offset_dict['yoff'] + row_index
+            for col_index in range(offset_dict['win_xsize']):
+                xs = offset_dict['xoff'] + col_index
 
                 # if this can be a seed pixel and hasn't already been
                 # calculated, put it on the stack
-                if seed_pixel and sediment_deposition_raster.get(
-                        seed_col, seed_row) == target_nodata:
-                    processing_stack.push(seed_row * n_cols + seed_col)
+                if (mfd_flow_direction_raster.is_local_high_point(xs, ys) and
+                        sediment_deposition_raster.get(xs, ys) == target_nodata):
```

Comment:
Sure, I changed all of these to `numpy.isclose`

Reply:
Uffda, that `numpy.isclose` is a python function, so it will be _super_ slow to call!  I would strongly discourage calling `numpy.isclose()` unless there's a C or C++ equivalent available. Instead, if there are different combinations of types we should be handling in our equality checks, the more direct solution might just be to adapt `_is_close()` to handle the types we're targeting.

Reply:
Good point @phargogh. When I used `numpy.isclose`, the test suite ran noticeably slower. I added a `is_close` cython function to `managed_raster.pxd` that can be used in all the other cython modules.